astropy/units/format/base.py

```diff
@@ -24,15 +26,96 @@ def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
 
     @classmethod
-    def parse(cls, s):
+    def _get_unit_name(cls, unit):
+        return unit.get_format_name(getattr(cls, "name", None))
+
+    @classmethod
+    def format_exponential_notation(cls, val, format_spec="g"):
         """
-        Convert a string to a unit object.
+        Formats a value in exponential notation.
+
+        Parameters
+        ----------
+        val : number
+            The value to be formatted
+
+        format_spec : str, optional
+            Format used to split up mantissa and exponent
+
+        Returns
+        -------
+        string : str
+            The value in exponential notation in a this class's format.
         """
-        raise NotImplementedError(f"Can not parse with {cls.__name__} format")
+        return format(val, format_spec)
+
+    @classmethod
+    def _format_superscript(cls, number):
+        return f"({number})" if "/" in number or "." in number else number
 
     @classmethod
-    def to_string(cls, u):
+    def _format_unit_power(cls, unit, power=1):
+        """Format the unit for this format class raised to the given power.
+
+        This is overridden in Latex where the name of the unit can depend on the power
+        (e.g., for degrees).
         """
-        Convert a unit object to a string.
+        name = cls._get_unit_name(unit)
+        if power != 1:
+            name += cls._format_superscript(utils.format_power(power))
+        return name
+
+    @classmethod
+    def _format_unit_list(cls, units):
+        return cls._space.join(
+            cls._format_unit_power(base_, power) for base_, power in units
+        )
+
+    @classmethod
+    def _format_fraction(cls, scale, nominator, denominator):
+        if cls._space in denominator:
+            denominator = f"({denominator})"
+        if scale and nominator == "1":
+            return f"{scale}/ {denominator}"
+        return f"{scale}{nominator} / {denominator}"
+
+    @classmethod
+    def to_string(cls, unit, inline=False):
+        if unit.scale == 1:
+            s = ""
+        else:
+            # Non-unity scale happens mostly for decomposed units.
+            # E.g., u.Ry.decompose() gives "2.17987e-18 kg m2 / s2".
+            s = cls.format_exponential_notation(unit.scale)
```

Comment:
Not sure this is an improvement...

```suggestion
        # Non-unity scale happens mostly for decomposed units.
        # E.g., u.Ry.decompose() gives "2.17987e-18 kg m2 / s2".
        s = cls.format_exponential_notation(unit.scale) if unit.scale != 1 else ""
```

Reply:
Think here the explicit stanza is better. I moved the comment outside, though, so that it is clear this part is about the scale.